catalog/rest/catalog-rest-endpoint/src/main/java/org/codice/ddf/endpoints/rest/RESTEndpoint.java

```diff
@@ -840,10 +879,12 @@ public Response addDocument(
                           createInfo.getMetacard())),
                   null);
           createResponse = catalogFramework.create(streamCreateRequest);
+          // TODO phil: the convention I've been following is that the response header contains the
+          // ID of the parent metacard, but at this point, I don't know which metacard is the
+          // parent.
+          id = createResponse.getCreatedMetacards().get(0).getId();
```

Comment:
Again we might need to check the response before using it

Reply:
added null checks